### PR TITLE
chore: rework dev setup

### PR DESCRIPTION
**Pull request by jooola**
- Merge configuration files into a single one
- Simplify the Makefile
- Merge the dependencies locations into setup.py
- Do not run pre-commit though tox
- Remove unused files or configuration
- Improve the CI definition
- Improve the python package metadata, by adding links and grouping related fields
- Add link to changelog in the python package and remove the changelog from the package description

### CODE REVIEW

**Comment by jooola** [comment]
This PR might be a bit aggressive, tell me if I should explain some changes and my plan.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (apricote) [inline .github/workflows/lint.yml:7]: I think we can just run this on all pull requests, when I have some complicated changes I sometimes like to use [Stacked PRs](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/), this workflow would not run for them.

This also applies to `.github/workflows/test.yml`

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,42 +1,29 @@
...
-
-.tests_template: &tests_template
-  before_script:
-    - pip install tox
-    - apk add build-base git
-  stage: test
-  script: tox
+default:
   tags:
     - hc-bladerunner
```
Comment (apricote) [inline .gitlab-ci.yml:3]: Could you change the tag to `cloud-integrations`? The `hc-bladerunner` runners are not available anymore

File: .github/workflows/lint.yml (hunk 1 above)
Comment (apricote) [reply .github/workflows/lint.yml]: Actuall, it was fine having it only run on pushes to `main`. If we have `push` & `pull_request` for all branches, pull requests will have double jobs running. This is already happening on some of our projects like https://github.com/hetznercloud/hcloud-go/pull/277/checks

Comment (jooola) [reply .github/workflows/lint.yml]: Alright, fixed it.

**Review by apricote — APPROVED** [review]

**Pull request merged**